### PR TITLE
Module Release 0.2.0

### PR DESCRIPTION
**Pull request by vhrychenko**
- Developer(s): @vhrychenko

- Ticket: https://spryker.atlassian.net/browse/ECO-2729

#### Please confirm

- [x] No new OS components - or they have been approved by the legal department

#### Documentation

- [x] Functional documentation provided or in progress?
- [x] Integration guide for projects provided or not needed?
- [x] Migration guides for all contained majors provided or not needed?

#### Release Table

   Module                | Release Type         | Constraint Updates         |
   :--------------------- | :------------------------ | :--------------------- |
   Ratepay  | minor                 |                       |

#### Release Notes

* FE of the module was refactored. Now it's compatible with suite projects.
* Created a new widget with Ratepay summary info.

-----------------------------------------

#### Module Ratepay

_**Patch:** Backwards-compatible bug fix_

Def of done (by the responsible developer):
- [x] All changes are backward-compatible. An outdated code is marked as deprecated.
- [x] The change is isolated and not mixed with any cleanups or other changes.
- [x] New and changed facade methods covered by functional tests. / Has nothing to test.
- [x] New and changed complex business logic is covered by unit or integration tests. / Has nothing to test.

### Improvements

- FE of the module was refactored. Now it's compatible with suite projects.
- Created a new widget with Ratepay summary info.

### CODE REVIEW

File: src/SprykerEco/Yves/Ratepay/Theme/default/partial/ratepay-summary-widget.twig [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    quoteTransfer: _widget.quoteTransfer,
+    view: _widget.view
+} %}
+
+{% block body %}
+    {% include '@Ratepay/partial/summary/' ~ data.view ~ '.twig' with {
+        ratepayPaymentTransfer: data.ratepayPaymentTransfer,
+        quoteTransfer: data.quoteTransfer
+    } %}
+
```
Comment (lockw1n) [inline src/SprykerEco/Yves/Ratepay/Theme/default/partial/ratepay-summary-widget.twig]: Please remove redundant line.

File: src/SprykerEco/Yves/Ratepay/Theme/default/partial/ratepay-summary-widget.twig [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    {% if data.ratepayPaymentTransfer is defined %}
+            <script>
+                var di = {t:'{{ data.ratepayPaymentTransfer.deviceFingerprint }}',v:'{{ data.ratepayPaymentTransfer.deviceIdentSId }}',l:'Checkout'}; </script>
+            <script type="text/javascript" src="//d.ratepay.com/{{ data.ratepayPaymentTransfer.deviceIdentSId }}/di.js"></script> <noscript><link rel="stylesheet" type="text/css" href="//d.ratepay.com/di.css?t={{ data.ratepayPaymentTransfer.deviceFingerprint }}&v={{ data.ratepayPaymentTransfer.deviceIdentSId }}&l=Checkout"></noscript>
+            <object type="application/x-shockwave-flash" data="//d.ratepay.com/{{ data.ratepayPaymentTransfer.deviceIdentSId }}/c.swf" width="0" height="0">
+                <param name="movie" value="//d.ratepay.com/{{ data.ratepayPaymentTransfer.deviceIdentSId }}/c.swf" />
+                <param name="flashvars" value="t={{ data.ratepayPaymentTransfer.deviceFingerprint }}&v={{ data.ratepayPaymentTransfer.deviceIdentSId }}"/><param name="AllowScriptAccess" value="always"/>
+            </object>
+    {% endif %}
+
```
Comment (lockw1n) [inline src/SprykerEco/Yves/Ratepay/Theme/default/partial/ratepay-summary-widget.twig]: Please remove redundant line.

File: src/SprykerEco/Yves/Ratepay/Theme/default/partial/ratepay-summary-widget.twig [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+{% endblock %}
+
```
Comment (lockw1n) [inline src/SprykerEco/Yves/Ratepay/Theme/default/partial/ratepay-summary-widget.twig]: Leave only one empty line at the end of the file.

File: src/SprykerEco/Yves/Ratepay/Widget/RatepaySummaryWidget.php [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        $this->addParameter(static::PARAMETER_VIEW_NAME, $this->getViewName($quoteTransfer));
+        $this->addParameter(static::PARAMETER_QUOTE_TRANSFER, $quoteTransfer);
+    }
+
+    /**
+     * @return string
+     */
+    public static function getName(): string
+    {
+        return 'RatepaySummaryWidget';
```
Comment (lockw1n) [inline src/SprykerEco/Yves/Ratepay/Widget/RatepaySummaryWidget.php]: Would be nice to have it as a constant.

File: src/SprykerEco/Yves/Ratepay/Widget/RatepaySummaryWidget.php [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    }
+
+    /**
+     * @return string
+     */
+    public static function getTemplate(): string
+    {
+        return '@Ratepay/partial/ratepay-summary-widget.twig';
```
Comment (lockw1n) [inline src/SprykerEco/Yves/Ratepay/Widget/RatepaySummaryWidget.php]: Would be nice to have it as a constant.

**Review by lockw1n — APPROVED** [review]

**Comment by lockw1n** [comment]
Everything's fine from my side.

File: src/SprykerEco/Yves/Ratepay/Theme/default/partial/ratepay-summary-widget.twig [hunk 6]
```diff
@@ -0,0 +1,24 @@
+{% extends template('widget') %}
```
Comment (tolerants) [inline src/SprykerEco/Yves/Ratepay/Theme/default/partial/ratepay-summary-widget.twig]: Please move this file to `Theme/default/templates`.

File: src/SprykerEco/Yves/Ratepay/Theme/default/partial/ratepay-summary-widget.twig [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+{% define data = {
+    ratepayPaymentTransfer: _widget.ratepayPaymentTransfer,
+    quoteTransfer: _widget.quoteTransfer,
+    view: _widget.view
+} %}
+
+{% block body %}
+    {% if data.ratepayPaymentTransfer is not null %}
+        {% include '@Ratepay/partial/summary/' ~ data.view ~ '.twig' with {
```
Comment (tolerants) [inline src/SprykerEco/Yves/Ratepay/Theme/default/partial/ratepay-summary-widget.twig]: We are not using partials. You have to create a view for every partial file you have and extend template.

**Comment by tolerants** [comment]
Why do you need empty files?

**Review by tolerants — CHANGES_REQUESTED** [review]

File: src/SprykerEco/Yves/Ratepay/Theme/default/partial/summary/elv.twig [hunk 8]
```diff
@@ -0,0 +1,12 @@
+<div class="box">
```
Comment (tolerants) [inline src/SprykerEco/Yves/Ratepay/Theme/default/partial/summary/elv.twig]: Please move into `Theme/default/views/summary-elv/summary-elv.twig`.

**Review by tolerants — APPROVED** [review]

**Pull request merged**